waspc/cli/Command/Info.hs

```diff
@@ -0,0 +1,74 @@
+module Command.Info
+  ( info,
+  )
+where
+
+import qualified Cli.Common
+import Cli.Terminal (title)
+import Command (Command)
+import Command.Common
+  ( findWaspProjectRootDirFromCwd,
+    waspSaysC,
+  )
+import Common (WaspProjectDir)
+import Control.Monad.IO.Class (liftIO)
+import Lib (findWaspFile)
+import qualified Parser
+import qualified Path as P
+import StrongPath (Abs, Dir, Path', toFilePath)
+import StrongPath.Operations
+import StrongPath.Path (toPathAbsDir)
+import System.Directory (doesFileExist, getFileSize)
+import Util.IO (listDirectoryDeep)
+import qualified Util.Terminal as Term
+import Wasp (Wasp, appName, getApp)
+
+info :: Command ()
+info =
+  do
+    waspDir <- findWaspProjectRootDirFromCwd
+    compileInfo <- liftIO $ readCompileInformation waspDir
+    projectSize <- liftIO $ readDirectorySizeMB waspDir
+    waspAstOrError <- liftIO $ parseWaspFile waspDir
+    case waspAstOrError of
+      Left err -> waspSaysC err
+      Right wasp -> do
+        waspSaysC $
+          unlines
+            [ "",
+              title "Project information",
+              printInfo
+                "Name"
+                (appName $ getApp wasp),
+              printInfo
+                "Last compile"
+                compileInfo,
+              printInfo
+                "Project size"
+                projectSize
+            ]
+
+printInfo :: String -> String -> String
+printInfo key value = Term.applyStyles [Term.Cyan] key ++ ": " <> Term.applyStyles [Term.White] value
+
+readDirectorySizeMB :: Path' Abs (Dir WaspProjectDir) -> IO String
+readDirectorySizeMB path = (++ " MB") . show . (`div` 1000000) . sum <$> (listDirectoryDeep (toPathAbsDir path) >>= mapM (getFileSize . P.toFilePath))
+
+readCompileInformation :: Path' Abs (Dir WaspProjectDir) -> IO String
+readCompileInformation waspDir = do
+  let dotWaspInfoFile = toFilePath $ waspDir </> Cli.Common.dotWaspDirInWaspProjectDir </> Cli.Common.generatedCodeDirInDotWaspDir </> Cli.Common.dotWaspInfoFileInGeneratedCodeDir
+  dotWaspInfoFileExists <- doesFileExist dotWaspInfoFile
+  if dotWaspInfoFileExists
+    then do readFile dotWaspInfoFile
```

Comment:
One tricky thing here is that, if we modify waspinfo file in the future to contain more information than just compile info, we will probably forget about this piece of code here and will still be printing all of it here and it will probably be a bunch of non-revelant information. But ok, I don't see a simple way to improve that right now. We could introduce some kind of abstraction over waspinfo file that implements a function that returns only compile info from its contents (for now that would be `id`), and then it will be more obvious when we change it that we have that assumption, but I wouldn't waste time on it at the moment.

Reply:
Oh yes I see, it makes total sense. Maybe expanding `.waspinfo` to some markup language to parse easily, like a yaml file with some keys. Or an easy format we can parse, assuming that `.waspinfo` will grow and some other info will be there and granular access will be needed.

The other solution that doesn't involves introducing new markup or formats of some sort can be using the modification date of `.waspinfo`. Making the assumption that, the last time that this file was modified, it was also the last time that it was compiled. 

However this can also mean that manual modification will also change compilation information, which is not optimal.

But yes, as you mentioned, I believe that an abstraction to not forget it, will be reasonable way to go at the moment. 👍

Reply:
That is a way to do it also! But as you said, we are making assumptions nobody else is touching it + that dotwaspinfo will be changed only on compile, which might not be true in the future, wasp might be storing additional information there.

I don't think we should be solving this now, just wanted to give heads up.